components/infobox/extensions/commons/age.lua

```diff
@@ -0,0 +1,235 @@
+
+	local formatString = self:_getFormatString()
+	local timestamp = self:getEarliestPossible()
+
+	if formatString then
+		local formatted = os.date(formatString, timestamp)
+		if not String.isEmpty(self.location) then
+			formatted = formatted .. '<br>' .. self.location
+		end
+
+		return formatted
+	end
+
+	return ''
+end
+
+function Date:makeIso()
+	if self.isEmpty or not self.isExact then
+		return ''
+	end
+
+	local year = self.year
+	local month = self.month
+	local day = self.day
+
+	return year .. '-' .. month .. '-' .. day
+end
+
+function Date:_getFormatString()
+	if self.year then
+		if self.month and self.day then
+			return '%B %e, %Y'
+		elseif self.month then
+			return '%B, %Y'
+		else -- Ignore day if we do not know the month
+			return '%Y'
+		end
+	elseif self.month and self.day then
+		return '%B %e'
+	elseif self.month then
+		return '%B'
+	end
+end
+
+function Date:getEarliestPossible()
+	return os.time({
+		year = self.year or _EPOCH_DATE.year,
+		month = self.month or _EPOCH_DATE.month,
+		day = self.day or _EPOCH_DATE.day,
+	})
+end
+
+function Date:getLatestPossible()
+	return os.time({
+		year = self.year or _CURRENT_YEAR,
+		month = self.month or _MONTH_DECEMBER,
+		day = self.day or _DEFAULT_DAYS_IN_MONTH[self.month or _MONTH_DECEMBER],
+	})
+end
+
+local BirthDate = Class.new(Date)
+local DeathDate = Class.new(Date)
+
+local Age = Class.new(
+	function(self, birthDate, deathDate)
+		self.birthDate = birthDate
+		self.deathDate = deathDate
+	end
+)
+
+function Age:calculate()
+	local endDate
+	if self.deathDate.isEmpty then
+		endDate = os.time()
+	else
+		endDate = self.deathDate.time
+	end
+
+	if self.birthDate.isExact and (self.deathDate.exact or self.deathDate.isEmpty) then
+		return self:_secondsToAge(os.difftime(endDate, self.birthDate.time))
+	elseif self.birthDate.year and (self.deathDate.year or self.deathDate.isEmpty) then
+		local minEndDate
+		local maxEndDate
+		if self.deathDate.isEmpty then
+			minEndDate = os.time()
+			maxEndDate = os.time()
+		else
+			minEndDate = self.deathDate:getEarliestPossible()
+			maxEndDate = self.deathDate:getLatestPossible()
+		end
+
+		local minAge = self:_secondsToAge(os.difftime(minEndDate, self.birthDate:getLatestPossible()))
+		local maxAge = self:_secondsToAge(os.difftime(maxEndDate, self.birthDate:getEarliestPossible()))
+
+		-- If our min and max age values are identical, then we can just display that value. Else,
+		-- we have to display a range of ages
+		if minAge == maxAge then
+			return minAge
+		else
+			return minAge .. '-' .. maxAge
+		end
+	end
+
+	return nil
+end
+
+function Age:makeDisplay()
+	local age = self:calculate()
+	local result = {
+		death = nil,
+		birth = nil,
+	}
+
+	if age ~= nil then
+		if not self.deathDate.isEmpty then
+			result.death = self.deathDate:makeDisplay() .. ' (aged ' .. age .. ')'
+			result.birth = self.birthDate:makeDisplay()
+		else
+			result.birth = self.birthDate:makeDisplay() .. ' (age ' .. age .. ')'
+		end
+	end
```

Comment:
```suggestion
	or
		result.birth = self:birthDate:makeDisplay()
		result.death = self:deathDate:makeDisplay()
	end
```
else lines 92-95 are useless and we would not be able to display dates without year (due to age only being calculated for dates with years)

Reply:
"or"? That's not a keyword? Not sure what you want to happen here

Reply:
suggesting to add the lines
```lua
	or
		result.birth = self:birthDate:makeDisplay()
		result.death = self:deathDate:makeDisplay()
```
before the `end`

so that we get display even if the age/year is unknown (allowing e.g. `January 5` as birthdate display)

Reply:
But "or" is not a thing?

Reply:
ah rofl brain fart on my side ... should have been `else`

Reply:
Done